src/javascript/jest.ts

```diff
@@ -817,6 +822,51 @@ export class Jest extends Component {
     this.testMatch.push(pattern);
   }
 
+  /**
+   * Build standard test match patterns for a directory.
+   * @param dirs The directories to add test matches for. Matches any folder if not specified or an empty array.
+   * @param fileExtensionPattern The file extension pattern to use. Defaults to "[jt]s?(x)".
+   */
+  public discoverTestMatchPatternsForDirs(
+    dirs?: string[],
+    fileExtensionPattern = "[jt]s?(x)"
+  ): void {
```

Comment:
I think we need one more iteration on the API. Something like this:
```suggestion
  public discoverTests(
    directories: string[],
    options?: DiscoverTestsOptions
  ): void {
  const patterns = options.fileExtensionPattern ?? "[jt]s?(x)";
```

with 

```
interface DiscoverTestsOptions {
    fileExtensionPattern?: String
}
```